packages/kotti-ui/source/kotti-filter/types.ts

```diff
@@ -0,0 +1,158 @@
+import { Kotti } from '../types'
+import { SpecifyRequiredProps } from '../types/utilities'
+
+export namespace KottiFilter {
+	export enum ButtonLinkType {
+		DANGER = 'danger',
+		PRIMARY = 'primary',
+	}
+
+	export enum FilterType {
+		BOOLEAN = 'BOOLEAN',
+		DATE_RANGE = 'DATE_RANGE',
+		MULTI_ENUM = 'MULTI_ENUM',
+		NUMBER = 'NUMBER',
+		SEARCH = 'SEARCH',
+		SINGLE_ENUM = 'SINGLE_ENUM',
+		STRING = 'STRING',
+	}
+
+	export namespace Operation {
+		export enum String {
+			CONTAINS = 'CONTAINS',
+		}
+
+		export enum Number {
+			EQUAL = 'EQUAL',
+			GREATER_THAN = 'GREATER_THAN',
+			GREATER_THAN_OR_EQUAL = 'GREATER_THAN_OR_EQUAL',
+			LESS_THAN = 'LESS_THAN',
+			LESS_THAN_OR_EQUAL = 'LESS_THAN_OR_EQUAL',
+		}
+
+		export enum Boolean {
+			EQUAL = 'EQUAL',
+		}
+
+		export enum DateRange {
+			IN = 'IN',
+		}
+
+		export enum SingleEnum {
+			EQUAL = 'EQUAL',
+		}
+
+		export enum MultiEnum {
+			ONE_OF = 'ONE_OF', // OR
+		}
+	}
+
+	export type Column = {
+		key: string
+		label: string
+	} & (
+		| {
+				type: FilterType.SEARCH
+				placeholder?: string | null
+		  }
+		| {
+				type: FilterType.STRING
+		  }
+		| {
+				type: FilterType.NUMBER
+		  }
+		| {
+				type: FilterType.BOOLEAN
+		  }
+		| {
+				type: FilterType.DATE_RANGE
+		  }
+		| {
+				type: FilterType.SINGLE_ENUM
+				options: Kotti.FieldSingleSelect.Props['options']
+		  }
+		| {
+				type: FilterType.MULTI_ENUM
+				options: Kotti.FieldMultiSelect.Props['options']
+		  }
+	)
+
+	export type FilterOperation =
+		| Operation.Boolean
+		| Operation.DateRange
+		| Operation.MultiEnum
+		| Operation.Number
+		| Operation.SingleEnum
+		| Operation.String
+
+	export type FilterValue =
+		| boolean
```

Comment:
Shouldn’t this be `Kotti.FieldToggle.Value`, `Kotti.FieldText.Value` and `Kotti.FieldNumber.value`?